io/js/src/main/scala/fs2/io/ioplatform.scala

```diff
@@ -126,7 +130,8 @@ private[fs2] trait ioplatform {
                 )
               }
             } >> go(tail)
-          case None => Pull.eval(F.delay(writable.end())) >> Pull.done
+          case None =>
+            Pull.eval(F.async_[Unit](cb => writable.end(() => cb(Right(()))))) >> Pull.done
```

Comment:
The `Pull.done` looks redundant here

Reply:
I was wondering that too! Still new to fs2, thanks :)